app/api/search.v2/routes.ts

```diff
@@ -0,0 +1,63 @@
+import { Application, Request, Response, NextFunction } from 'express';
+//@ts-ignore
+import queryTypes from 'query-types';
+
+import { elastic } from 'api/search/elastic';
+import validateRequest from 'api/utils/validateRequest';
+import { SearchQuerySchema } from 'shared/types/SearchQuerySchema';
+import { SearchQuery } from 'shared/types/SearchQueryType';
+
+import { buildQuery } from './buildQuery';
+
+interface UwaziResponse {
+  data: any;
+  links?: {
+    self: string;
+    first?: string | null;
+  };
+}
+
+interface UwaziReq<T> extends Request {
+  query: T;
+}
+
+type UwaziRes = Omit<Response, 'json'> & { json(data: UwaziResponse): Response };
+
+const captureError = (
+  callback: (req: Request, res: Response, next: NextFunction) => Promise<void>
+) => async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    await callback(req, res, next);
+  } catch (e) {
+    next(e);
+  }
+};
+
+const searchRoutes = (app: Application) => {
+  app.get(
+    '/api/v2/entities',
+    queryTypes.middleware(),
+    validateRequest({
+      query: SearchQuerySchema,
```

Comment:
this validation has no tests, and i think is wrong it should be something like

Reply:
added test and fixed